planemo/workflow_lint.py

```diff
@@ -133,13 +201,37 @@ def _lint_case(path, test_case, lint_context):
             found_valid_expectation = True
         # TODO: validate structure of test expectations
 
+        assertion_definitions = test_case.output_expectations[test_output_id].get("asserts")
+        if not _check_test_assertions(lint_context, assertion_definitions):
+            test_valid = False
+
     if not found_valid_expectation:
         lint_context.warn("Found no valid test expectations for workflow test")
         test_valid = False
 
     return test_valid
 
 
+def _check_test_assertions(lint_context, assertion_definitions):
+    # we are already in Python, not XML, so it is simpler to lint assertions by checking against the
+    # Python functions directly, rather than checking against galaxy.xsd as for tool linting
+    assertions_valid = True
+    if assertion_definitions:
+        for module in asserts.assertion_modules:
+            for function_name in dir(module):
+                if function_name.split('assert_')[-1] in assertion_definitions:
+                    signature = inspect.signature(module.__dict__[function_name])
+                    try:
+                        # try mapping the function with the attributes supplied and check for TypeError
+                        signature.bind('', **assertion_definitions[function_name.split('assert_')[-1]])
+                    except AssertionError:
+                        pass
+                    except TypeError as e:
+                        lint_context.error(f'Invalid assertion in tests: {function_name} {str(e)}')
+                        assertions_valid = False
```

Comment:
```suggestion
        for function_name, function in asserts.assertion_functions.items():
            if function_name.split('assert_')[-1] in assertion_definitions:
                signature = inspect.signature(function)
                try:
                    # try mapping the function with the attributes supplied and check for TypeError
                    signature.bind('', **assertion_definitions[function_name.split('assert_')[-1]])
                except AssertionError:
                    pass
                except TypeError as e:
                    lint_context.error(f'Invalid assertion in tests: {function_name} {str(e)}')
                    assertions_valid = False
```
When https://github.com/galaxyproject/galaxy/pull/12528 is incorporated into a published version of `galaxy-tool-util` this change will be needed.